apps/wallet/src/ui/app/shared/forms/controls/PasswordInput.tsx

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { EyeOpen16, EyeClose16 } from '@mysten/icons';
+import { forwardRef, useState, type ComponentProps } from 'react';
+import { Input } from './Input';
+import { ButtonOrLink } from '../../utils/ButtonOrLink';
+
+type PasswordInputProps = {
+	name: string;
+} & Omit<ComponentProps<'input'>, 'className' | 'type' | 'name' | 'ref'>;
+
+export const PasswordInput = forwardRef<HTMLInputElement, PasswordInputProps>(
+	({ placeholder, ...props }, forwardedRef) => {
+		const [passwordShown, setPasswordShown] = useState(false);
+		const IconComponent = passwordShown ? EyeOpen16 : EyeClose16;
+
+		return (
+			<div className="flex w-full relative items-center">
+				<Input
+					type={passwordShown ? 'text' : 'password'}
+					placeholder="Password"
+					ref={forwardedRef}
+					{...props}
```

Comment:
I think this is breaking the show/hide password button as `type="password"` is being passed through from `ProtectAccountForm`

Reply:
ah good catch, I fixed this by spreading the props first then overriding the type